src/Avalonia.Build.Tasks/XamlCompilerTaskExecutor.cs

```diff
@@ -62,21 +66,51 @@ public CompileResult(bool success, bool writtenFile = false)
 
             if (patchCom)
                 ComInteropHelper.PatchAssembly(asm, typeSystem);
-            
+
             var writerParameters = new WriterParameters { WriteSymbols = asm.MainModule.HasSymbols };
             if (!string.IsNullOrWhiteSpace(strongNameKey))
                 writerParameters.StrongNameKeyBlob = File.ReadAllBytes(strongNameKey);
 
             asm.Write(output, writerParameters);
 
             return new CompileResult(true, true);
-            
+
         }
-        
+
         static bool? CompileCore(IBuildEngine engine, CecilTypeSystem typeSystem,
             string projectDirectory, bool verifyIl, 
-            MessageImportance logImportance)
+            MessageImportance logImportance
+            , bool debuggerLaunch = false)
         {
+            if (debuggerLaunch)
+            {
+                // According this https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debugger.launch?view=net-6.0#remarks
+                // documentation, on not windows platform Debugger.Launch() always return true without running a debugger.
+                if (System.Diagnostics.Debugger.Launch())
+                {
+                    // Set timeout at 1 minut.
+                    var time = new System.Diagnostics.Stopwatch();
+                    var timeout = TimeSpan.FromMinutes(1);
+                    time.Start();
+
+                    // wait for the debugger to be attacked or timeout.
+                    while (!System.Diagnostics.Debugger.IsAttached && time.Elapsed < timeout)
+                    {
+                        engine.LogMessage($"Wating attach debugger. Elapsed {time.Elapsed}...", MessageImportance.Low);
```

Comment:
Since we are already doing that, change importance to High (so it's always visible) and print the current PID as well.